darkmoon/api/v1/metadata/views.py

```diff
@@ -367,22 +414,27 @@ async def hash_comparison(
         }
 
         results = await collection.find(search_query).to_list(length=length)
-        li = [MetadataEntity.parse_obj(item) for item in results]
+        li: list[MetadataEntity] = []
+        for item in results:
+            if "header_info" in item:
```

Comment:
I'm seeing this logic check a lot in this code, which indicates to me we need to refactor this into a single location. While you could just write a function that takes in a blob of data and returns either an ExeMetadataEntity or DocMetadataEntity, could you also just refactor the models in to a tagged union and let Pydantic do the validation for you. https://docs.pydantic.dev/1.10/usage/types/#discriminated-unions-aka-tagged-unions